### PR TITLE
8331862: Remove split relocation info implementation

### PR DESCRIPTION
**Pull request by vnkozlov**
[Split relocation info](https://github.com/openjdk/jdk/blob/master/src/hotspot/share/code/relocInfo.hpp#L225) was used only for SPARC. Non of current OpenJDK platforms use it.

Tested tier1-3,stress,xcomp.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8331862](https://bugs.openjdk.org/browse/JDK-8331862): Remove split relocation info implementation (**Enhancement** - P4)

### Reviewers
 * [Dean Long](https://openjdk.org/census#dlong) (@dean-long - **Reviewer**) ⚠️ Review applies to [64c9e66b](https://git.openjdk.org/jdk/pull/19126/files/64c9e66b01adc25632610fd4418cd0352c1af04d)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19126/head:pull/19126` \
`$ git checkout pull/19126`

Update a local copy of the PR: \
`$ git checkout pull/19126` \
`$ git pull https://git.openjdk.org/jdk.git pull/19126/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19126`

View PR using the GUI difftool: \
`$ git pr show -t 19126`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19126.diff">https://git.openjdk.org/jdk/pull/19126.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19126#issuecomment-2098841493)

### CODE REVIEW

File: src/hotspot/cpu/s390/assembler_s390.cpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016, 2021, Oracle and/or its affiliates. All rights reserved.
```
Comment (dean-long) [inline src/hotspot/cpu/s390/assembler_s390.cpp:2]: No changes to this file?

File: src/hotspot/share/code/relocInfo.hpp [hunk 2]
```diff
@@ -130,11 +130,7 @@ class nmethod;
 //   Instruction types: memory (load), set (load address)
 //   Data:  []       an oop stored in 4 bytes of instruction
 //          [n]      n is the index of an oop in the CodeBlob's oop pool
-//          [[N]n l] and l is a byte offset to be applied to the oop
-//          [Nn Ll]  both index and offset may be 32 bits if necessary
-//   Here is a special hack, used only by the old compiler:
-//          [[N]n 00] the value is the __address__ of the nth oop in the pool
-//   (Note that the offset allows optimal references to class variables.)
+//          [Nn]     index may be 32 bits if necessary
```
Comment (dean-long) [inline src/hotspot/share/code/relocInfo.hpp:133]: Lines 132 and 133 could be combined into something like:
```
// [[N]n]  index of an oop in the CodeBlob's oop pool
```
which seems consistent with other descriptions.

**Review by dean-long — APPROVED** [review]

File: src/hotspot/cpu/s390/assembler_s390.cpp (hunk 1 above)
Comment (vnkozlov) [reply src/hotspot/cpu/s390/assembler_s390.cpp]: Accidental change. Reverted.

File: src/hotspot/share/code/relocInfo.hpp (hunk 2 above)
Comment (vnkozlov) [reply src/hotspot/share/code/relocInfo.hpp]: Okay.

**Review by vnkozlov — COMMENTED** [review]
Thank you, @dean-long, for review. I addressed all your comments.

**Comment by vnkozlov** [comment]
@TheRealMDoerr, @RealFYang, @offamitkumar, @bulasevich 
I need your help with testing this on your platforms, at least tier1. GHA does some cross compilation but not testing.

Thanks!

**Comment by offamitkumar** [comment]
Result Looks good on s390x. I  ran `tier1` tests on `fastdebug-vm`.

**Comment by RealFYang** [comment]
Also performed `tier1` tests on linux-riscv64 platform. Result looks good.

**Comment by vnkozlov** [comment]
Thank you, Amit and Fei, for testing.

**Comment by TheRealMDoerr** [comment]
tier1 and many more tests have also passed on PPC64.

**Comment by vnkozlov** [comment]
Thank you, Martin

**Pull request closed without merging**